### DISCUSSION TITLE
Cant receive notification sound on background state on IOS

### DISCUSSION THREAD

**Discussion by Sargnec** [invertase/react-native-firebase#4633] [Q&A] [answered]
We are receiving notification but without sound. And

```
setBackgroundMessageHandler(){
        console.log('setBackgroundMessageHandler listener');
        return firebase.messaging().setBackgroundMessageHandler(async remoteMessage => {
            console.log('setBackgroundMessageHandler received',remoteMessage);
            console.log('Message handled in the background!', remoteMessage);
        });
}
```

this is not logging anything in terminal. What can I do about this?

**Comment by mikehardy** [comment, marked as answer]
Hi there!
When troubleshooting cloud messages, you should always post the message from the command line using hand-crafted JSON so you have reproducible tests, and you should include the JSON here so we know what the client is receiving.
Also there is no mention of platform or versions or anything - this isn't a discussion, it's an issue, and it should have been filed as an issue with those details (as requested in our issue template so we can troubleshoot)

None of that was done, so this is only a guess but:
- you're not posting a data-only message, so the SDK is handling the notification part of the payload internally and your handler is not triggered
- your notification doesn't have the contents you think in the JSON payload so the sound isn't found (perhaps `adb logcat` will illuminate this, but only on android)
- your notification channel doesn't include sound permissions (unknown, no JSON payload, no App.js to reproduce that would show channel construction, no indication of permissions, android only)

Hope this helps

**Reply by Sargnec** [reply]
```
{
    "to":"...",
    "data":{},
    "notification":{
        "title":"title",
        "body":"Test deneme 1 2",
        "sound" : "default"
    },
    "contentAvailable":true,
    "priority":"high"
}
```
Sending sound" : "default" in notification solved our problem
